src/tpm2/RuntimeProfile.c

```diff
@@ -0,0 +1,757 @@
+/********************************************************************************/
+/*										*/
+/*			       Runtime Profile 					*/
+/*			     Written by Stefan Berger				*/
+/*		       IBM Thomas J. Watson Research Center			*/
+/*										*/
+/*  Licenses and Notices							*/
+/*										*/
+/*  (c) Copyright IBM Corporation, 2022						*/
+/*										*/
+/* All rights reserved.								*/
+/*										*/
+/* Redistribution and use in source and binary forms, with or without		*/
+/* modification, are permitted provided that the following conditions are	*/
+/* met:										*/
+/*										*/
+/* Redistributions of source code must retain the above copyright notice,	*/
+/* this list of conditions and the following disclaimer.			*/
+/*										*/
+/* Redistributions in binary form must reproduce the above copyright		*/
+/* notice, this list of conditions and the following disclaimer in the		*/
+/* documentation and/or other materials provided with the distribution.		*/
+/*										*/
+/* Neither the names of the IBM Corporation nor the names of its		*/
+/* contributors may be used to endorse or promote products derived from		*/
+/* this software without specific prior written permission.			*/
+/*										*/
+/* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS		*/
+/* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT		*/
+/* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR	*/
+/* A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT		*/
+/* HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,	*/
+/* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT		*/
+/* LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,	*/
+/* DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY	*/
+/* THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT		*/
+/* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE	*/
+/* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.		*/
+/*										*/
+/********************************************************************************/
+
+#define _GNU_SOURCE
+#include <errno.h>
+#include <stdio.h>
+#include <regex.h>
+
+#include "Tpm.h"
+#include "tpm_library_intern.h"
+
+struct RuntimeProfile g_RuntimeProfile;
+
+static const struct RuntimeProfileDesc {
+    const char *name;
+    const char *commandsProfile;
+    const char *algorithmsProfile;
+    /* StateFormatLevel drives the format the TPM's state is written in and
+     * how it is read.
+     * Once a version of libtpms is released this field must never change afterwards
+     * so that backwards compatibility for reading the state can be maintained.
+     * This basically locks the name of the profile to the stateFormatLevel.
+     */
+    unsigned int stateFormatLevel;
```

Comment:
Can the stateFormatLevel be determined from the keysizes, types of algorithms, and commands enabled in a profile and be be removed from here? Will increasing the stateFormatLevel always be driven by new keysize, new algorithms, or new commands being enabled? (How to predict the future??)